patroni/dcs/consul.py

```diff
@@ -135,6 +135,9 @@ def connect(self, *args, **kwargs):
             kwargs['token'] = self._token
         return HTTPClient(**kwargs)
 
+    def set_token(self, token):
+        self._token = token
```

Comment:
`base.Consul` has `self.token`, which also must be updated. We created a duplicate `self._token` here because the connect method is called earlier than `self.token` is defined.

Or maybe it makes sense to rename `self._token` to `self.token`? IMHO it will not hurt.